connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -81,7 +81,26 @@ class SparkConnectPlanner(plan: proto.Relation, session: SparkSession) {
         } else {
           child
         }
-      case _ => throw InvalidPlanInput()
+      case proto.Read.ReadTypeCase.FILE_DATA_SOURCE =>
+        val reader = session.read
+        if (rel.getFileDataSource.getFormat != null) {
```

Comment:
I need to double check if there is a better way to check if a Java proto filed is not populated. I didn't find a `hasXXX` method in Intellij code auto-complete list.

Reply:
BTW I found that proto3 does not support `required` on proto fields anymore so we cannot rely on that.